aiida/engine/processes/calcjobs/tasks.py

```diff
@@ -11,19 +11,27 @@
 import functools
 import logging
 import tempfile
+from typing import Any, Callable, Optional, TYPE_CHECKING
 
 import plumpy
+import plumpy.process_states
+import plumpy.futures
 
 from aiida.common.datastructures import CalcJobState
 from aiida.common.exceptions import FeatureNotAvailable, TransportTaskException
 from aiida.common.folders import SandboxFolder
 from aiida.engine.daemon import execmanager
-from aiida.engine.utils import exponential_backoff_retry, interruptable_task
+from aiida.engine.transports import TransportQueue
+from aiida.engine.utils import exponential_backoff_retry, interruptable_task, InterruptableFuture
 from aiida.schedulers.datastructures import JobState
 from aiida.manage.configuration import get_config_option
 
 from ..process import ProcessState
 
+if TYPE_CHECKING:
+    from aiida.orm.nodes.process.calculation.calcjob import CalcJobNode
+    from .calcjob import CalcJob
```

Comment:
Is this just for efficiency? I don't think this could actually cause problems like cyclic dependencies, could it? Wouldn't it just be better to import it and use `CalcJobNode` for typing instead of `'CalcJobNode'`. Not sure what the standard is here for `mypy`.

Reply:
yeh just for cyclic dependencies. I can't remember if I specifically tested this one, but generally erred on the side of caution

Reply:
@sphuber do you think then that all aiida.orm imports in aiida/engine should be at the top of the file, since at present there is a number within functions/methods.

Note, I checked in `aiida/orm` and there is no `aiida.engine` imports except inside `aiida.orm.querybuilder.get_querybuilder_classifiers_from_cls` and inside `aiida.orm.utils.links.LinkManager.nested`

Reply:
I think indeed that `orm` should not depend on `engine`. So it should be possible to move all imports of `aiida.orm` in `aiida.engine` to the top and not have circular import problems. And unless this is a well-accepted standard practice in `mypy` to use the `if TYPE_CHECKING` guard if you only import something for typing, then I would get rid of it and just import it normally. I think the overhead should be minimal and it is less clutter

Reply:
sounds good to me 👍

Reply:
done (498dd02)